llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8301,6 +8305,9 @@ VPRecipeBuilder::tryToCreatePartialReduction(VPInstruction *Reduction,
          "all accumulators in chain must have same scale factor");
 
   auto *ReductionI = Reduction->getUnderlyingInstr();
+  if (Reduction->getOpcode() == Instruction::FAdd &&
+      (!ReductionI->hasAllowReassoc() || !ReductionI->hasAllowContract()))
+    return nullptr;
```

Comment:
Question: Can this be an assert due to the cost model changes, or is this exit still reachable?